src/Crdt.Core/HybridDateTime.cs

```diff
@@ -0,0 +1,94 @@
+﻿
+namespace Crdt.Core;
+
+/// <summary>
+/// Hybrid date time represents a hybrid logical clock that combines a datetime and a counter.
+/// <a href="https://martinfowler.com/articles/patterns-of-distributed-systems/hybrid-clock.html">read more here</a>
+/// It's use is primarily to avoid conflicts due to a users clock getting reset, we still want their changes to come
+/// after the most recent change they made.
+/// </summary>
+public record HybridDateTime : IComparable<HybridDateTime>
+{
+    private HybridDateTime()
+    {
+    }
+    public HybridDateTime(DateTimeOffset dateTime, long counter)
+    {
+        DateTime = dateTime;
+        Counter = counter;
+    }
+
+    public static HybridDateTime ForTestingNow => new(DateTimeOffset.UtcNow, 0);
```

Comment:
An alternative to putting test code into the class is to have this code get its dates from a provider like we do in some of our core library code. See SIL.Core/Providers/DateTimeProvider.cs

Reply:
I removed this property in favor of making times in test code directly where relevant